examples/ChatQnA/deploy/k8s_getting_started.md

```diff
@@ -0,0 +1,90 @@
+|`kubectl describe pod <pod-name>` | Provides detailed information about a specific pod, including its            current state, recent events, and configuration details.                            |
+|`kubectl delete deployments --all` |             Deletes all deployments in the current namespace, which effectively removes all the managed pods and associated resources.                |
+|`kubectl get pods -o wide`         |              Retrieves a detailed list of all pods in the current namespace, including additional information like IP addresses and the nodes they are running on.               |
+|`kubectl logs <pod-name>`         |         Fetches the logs generated by a container in a specific pod, useful for debugging and monitoring application behavior.                    |
+|`kubectl get svc`                  |            Lists all services in the current namespace, providing a quick overview of the network services and their status.
+
+#### Create and Set Namespace
+A Kubernetes namespace is a logical division within a cluster that is used to isolate different environments, teams, or projects, allowing for finer control over resources and access management. To create a namespace called `chatqa`, use:
+```bash
+kubectl create ns chatqa
+```
+When deploying resources (like pods, services, etc.) into your specific namespace, use the `--namespace` flag with `kubectl` commands, or specify the namespace in your resource configuration files.
+
+To deploy a pod in the `chatqa` namespace:
+```bash
+kubectl apply -f your-pod-config.yaml --namespace=chatqa
+```
+If you want to avoid specifying the namespace with every command, you can set the default namespace for your current context:
+```bash
+kubectl config set-context --current --namespace=chatqa
+```
+
+### Using Helm Charts to Deploy
+
+**What is Helm?** Helm is a package manager for Kubernetes, similar to how apt is for Ubuntu. It simplifies deploying and managing Kubernetes applications through Helm charts, which are packages of pre-configured Kubernetes resources.
+
+#### Key Components of a Helm Chart
+
+| Component         |Description                                                                                                                                             |
+| ---               | ---                                                                                                                                                    |
+| `Chart.yaml`      | This file contains metadata about the chart such as name, version, and description.                                                                    |
+| `values.yaml`     | Stores configuration values that can be customized depending on the deployment environment. These values override defaults set in the chart templates. |
+| `deployment.yaml` | Part of the templates directory, this file describes how the Kubernetes resources should be deployed, such as Pods and Services.                       |
```

Comment:
Incorrect.  Typically template directory has one file per k8s object, with object type being used as the file name.

Better to just mention template dir, and maybe have pointer to Helm docs: https://helm.sh/docs/chart_best_practices/templates/

Reply:
@eero-t I can change it to:

| `templates/ ` Directory | Contains YAML templates for Kubernetes objects, typically one file per object type (e.g., deployment.yaml for Deployments, service.yaml for Services). For more details, refer to the [Helm Templates Best Practices](https://helm.sh/docs/chart_best_practices/templates/).